pkg/controller/kogitodataindex/kogitodataindex_controller.go

```diff
@@ -175,6 +175,14 @@ func (r *ReconcileKogitoDataIndex) onDeploymentCreate(cli *client.Client, deploy
 		log.Warnf("No container definition for service %s. Skipping applying custom Data Index deployment configuration", kogitoService.GetName())
 	}
 
+	dataIndex := kogitoService.(*appv1alpha1.KogitoDataIndex)
+	if len(dataIndex.Spec.KogitoDataIndexPersistenceProvider) <= 0 || (dataIndex.Spec.KogitoDataIndexPersistenceProvider != appv1alpha1.InfinispanPersistenceProvider &&
+		dataIndex.Spec.KogitoDataIndexPersistenceProvider != appv1alpha1.MongoDBPersistenceProvider) {
```

Comment:
Having these values into a map would be easier to understand, something like:

```go
if _, ok := providers[dataIndex.Spec.KogitoDataIndexPersistenceProvider]; !ok {
    // invalid
}
```

Reply:
ok